### PR TITLE
Move more logic from distributive role into packages

### PR DESCRIPTION
**Pull request by siddharthist**
Should replace all the non-package logic in this role: https://github.com/CiscoCloud/mantl/blob/5531e07bebfe72645c32b2ffa6cb72d452fae1cb/roles/distributive/tasks/main.yml

* template consul configurations inside packages
* reload consul after installation
* set up docker permissions for consul user

creates new subdirectories of distributive: `checklists/` and `consul-configs/` for the two different kinds of packages. 

Note that all of the `scripts` sections are identical. Only the names change between packages and the variables they provide to the template "health-check-config.json.tmpl
". 

These have been tested successfully on mantl.

### CODE REVIEW

**Comment by stevendborrelli** [comment]
Consul doesn't need to be restarted when health check .json files are added/removed. A `consul reload` is much less likely to destabilize the cluster.

File: distributive/consul-configs/collectd/spec.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+scripts:
+  before-install: |
+      mkdir /etc/consul || true
+      chown -R consul:consul /etc/consul
+      # Allow Consul to run docker commands
+      echo 'consul ALL=(ALL)    NOPASSWD:   /bin/docker' > /etc/sudoers.d/consul-docker-auth
+      chown root:root '/etc/sudoers.d/consul-docker-auth'
+      visudo -c -f '/etc/sudoers.d/consul-docker-auth'
+      # Remove previous (deprecated) check
+      rm '/etc/distributive.d/{{variable "current_role"}}.json' || true
```
Comment (stevendborrelli) [inline distributive/consul-configs/collectd/spec.yml]: This pattern could be `rm -f` instead of `rm || true`

**Comment by siddharthist** [comment]
@stevendborrelli Thanks for the feedback! It's all changed, I'll test again with this configuration ASAP.

**Comment by siddharthist** [comment]
Closing due to lack of interest/added complexity.

**Pull request closed without merging**